src/backtick/core.clj

```diff
@@ -8,10 +8,16 @@
   (:gen-class))
 
 (defn register [name f]
-  (swap! engine/workers assoc name f))
+  ;; We store both the name => fn mapping (needed by submit-worker)
+  ;; and fn => name mapping (needed by schedule)
+  (swap! engine/workers assoc
+         name f
+         f name))
```

Comment:
You need to have the string the in the database somewhere.  If there was a way to turn a fn object to a canonical string (that worked across JVM instances) we wouldn't need the string part.  Take a look at `submit-worker` and you can see where we pull the fn name out the db.